### ISSUE TITLE
Immutability

### ISSUE THREAD

**Issue by finnhodgkin** [fac-13/SBSHS#28]
I find code easier to reason about if values aren't changed on the fly. For example in `Carousel.js`:

```js
let index = this.state.activeIndex;
let { slides } = this.props;
let slidesLength = slides.length;

if (index < 1) {
  index = slidesLength;
}

--index;
```
In the code above index changes three times. In this tiny example that's easy but as things grow in complexity it can become hard to keep track of the mutating values. 

As an example the complete section the code above was taken from could be refactored to something like this, where activeIndex is one thing in one place:
```js
goToPrevSlide = e => {
  e.preventDefault();

  const currentIndex = this.state.activeIndex;
  const slidesLength = this.props.slides.length
  // The desctructuring in the first example was redundant - it just added an extra line and a variable.

  const activeIndex = currentIndex < 1
    ? slidesLength - 1
    : currentIndex - 1;

  this.setState({ activeIndex });
};

goToNextSlide = e => {
  e.preventDefault();

  const currentIndex = this.state.activeIndex;
  const slidesLength = this.props.slides.length;

  const activeIndex = currentIndex >= slidesLength - 1
    ? 0 
    : currentIndex + 1;

  this.setState({ activeIndex });
};  
```
It's a tiny example to illustrate a bigger point (and really doesn't do a very good job...), but it can become a much larger problem on bigger projects. 

Basically just try to use `let` and `var` as little as possible.
